### PR TITLE
HDRP public API doc

### PR DESCRIPTION
**Pull request by JulienIgnace-Unity**
Updated missing doc in HDRP and HDRP config packages

### CODE REVIEW

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 1]
```diff
@@ -4,40 +4,67 @@
 
 namespace UnityEngine.Rendering.HighDefinition
 {
+    /// <summary>
+    /// Probe Volume Evaluation Mode.
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:8]: Options for the mode HDRP uses to evaluate Probe Volumes.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 2]
```diff
@@ -4,40 +4,67 @@
...
+    /// </summary>
     [GenerateHLSL(PackingRules.Exact)]
     public enum ProbeVolumesEvaluationModes
     {
+        /// <summary>Probe volumes are disabled.</summary>
         Disabled = 0,
+        /// <summary>Probe volumes are evaluated in the light loop.</summary>
         LightLoop = 1,
+        /// <summary>Probe volumes are evaluated in the material pass.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:17]: Evaluates probe volumes in the material pass.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 3]
```diff
@@ -4,40 +4,67 @@
...
 {
+    /// <summary>
+    /// Probe Volume Evaluation Mode.
+    /// </summary>
     [GenerateHLSL(PackingRules.Exact)]
     public enum ProbeVolumesEvaluationModes
     {
+        /// <summary>Probe volumes are disabled.</summary>
         Disabled = 0,
+        /// <summary>Probe volumes are evaluated in the light loop.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:15]: Evaluates probe volumes in the light loop.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 4]
```diff
@@ -4,40 +4,67 @@
 
 namespace UnityEngine.Rendering.HighDefinition
 {
+    /// <summary>
+    /// Probe Volume Evaluation Mode.
+    /// </summary>
     [GenerateHLSL(PackingRules.Exact)]
     public enum ProbeVolumesEvaluationModes
     {
+        /// <summary>Probe volumes are disabled.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:13]: Disables probe volumes.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs (hunk 3 above)
Comment (JordanL8) [reply com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:15]: Are there any remarks to add on the benefits of running probes in the light loop or in the material pass?

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 5]
```diff
@@ -4,40 +4,67 @@
...
         Disabled = 0,
+        /// <summary>Probe volumes are evaluated in the light loop.</summary>
         LightLoop = 1,
+        /// <summary>Probe volumes are evaluated in the material pass.</summary>
         MaterialPass = 2,
     }
 
+    /// <summary>
+    /// Encoding of Probe Volumes.
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:22]: Options for the method HDRP uses to encode probe volumes.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 6]
```diff
@@ -4,40 +4,67 @@
...
+    /// </summary>
     [GenerateHLSL(PackingRules.Exact)]
     public enum ProbeVolumesEncodingModes
     {
+        /// <summary>Encoded using L0 spherical harmonics.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:27]: Encodes probe volumes using L0 spherical harmonics.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 7]
```diff
@@ -4,40 +4,67 @@
...
         SphericalHarmonicsL0 = 0,
+        /// <summary>Encoded using L1 spherical harmonics.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:29]: Uses L1 spherical harmonics to encode probe volumes.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 8]
```diff
@@ -4,40 +4,67 @@
...
         SphericalHarmonicsL1 = 1,
+        /// <summary>Encoded using L2 spherical harmonics.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:31]: Uses L2 spherical harmonics to encode probe volumes.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs (hunk 6 above)
Comment (JordanL8) [reply com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:27]: What are these spherical harmonics and what are the benefits of using different levels?

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 9]
```diff
@@ -4,40 +4,67 @@
...
         SphericalHarmonicsL2 = 2
     }
 
+    /// <summary>
+    /// Probe Volume bilateral filtering mode.
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:36]: Options for the mode HDRP uses for probe volume bilateral filtering.

Comment (JordanL8) [reply com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:36]: Originally thought:
Options for the mode HDRP uses to bilaterally filter probe volumes.
but I'm not sure that conveys the same meaning. If you think it does, use this instead,

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 10]
```diff
@@ -4,40 +4,67 @@
...
+    /// </summary>
     [GenerateHLSL(PackingRules.Exact)]
     public enum ProbeVolumesBilateralFilteringModes
     {
+        /// <summary>Bilateral filtering is disabled.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:41]: Disables bilateral filtering.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 11]
```diff
@@ -4,40 +4,67 @@
...
         Disabled = 0,
+        /// <summary>Bilateral filtering using validity.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:43]: What is validity in this context?

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 12]
```diff
@@ -4,40 +4,67 @@
...
         Validity = 1,
+        /// <summary>Bilateral filtering using octahedral depth.</summary>
         OctahedralDepth = 2
     }
 
+    /// <summary>
+    /// Project wide shader configuration options.
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:50]: Project-wide shader configuration options.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 13]
```diff
@@ -4,40 +4,67 @@
...
+    /// This enum will generate the proper shader defines.
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:51]: This enum generates the correct shader defines for each option.

Move this into a /// <remarks> section.

Also, add a ///<seealso cref=""/> and point the cref to the property that uses this enum to define the actual shader options. Can you do this for the other enums above too please?

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 14]
```diff
@@ -4,40 +4,67 @@
...
+    /// </summary>
     [GenerateHLSL(PackingRules.Exact)]
     public enum ShaderOptions
     {
-        ColoredShadow = 1, // Allow to defined if colored shadow are supported in shaders or not
-        CameraRelativeRendering = 1, // Rendering sets the origin of the world to the position of the primary (scene view) camera
+        /// <summary>Support for colored shadows.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:56]: Supports colored shadows in shaders.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 15]
```diff
@@ -4,40 +4,67 @@
...
+        ColoredShadow = 1,
+        /// <summary>Use camera relative rendering to enhance precision.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:58]: Uses [camera relative rendering](../manual/Camera-Relative-Rendering.md) to enhance precision.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 16]
```diff
@@ -4,40 +4,67 @@
...
+        CameraRelativeRendering = 1,
+        /// <summary>Use pre-exposition to enhance color precision.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:60]: Uses pre-exposition to enhance color precision.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 17]
```diff
@@ -4,40 +4,67 @@
...
         PreExposition = 1,
-        PrecomputedAtmosphericAttenuation = 0, // Precomputes atmospheric attenuation for the directional light on the CPU, which makes it independent from the fragment's position, which is faster but wrong
+        /// <summary>Precomputes atmospheric attenuation for the directional light on the CPU, which makes it independent from the fragment's position, which is faster but wrong.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:62]: Precomputes atmospheric attenuation for the directional light on the CPU. This makes it independent from the fragment's position, which increases performance but reduces accuracy.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 18]
```diff
@@ -51,32 +78,55 @@ public enum ShaderOptions
         // Edit->Render Pipeline->Generate Shader Includes
         // Probe Volumes feature must also be enabled inside of your HDRenderPipelineAsset.
         // Also uncomment in the HDRP package all ".../Experimental/Probe Volume" menu
+
+        /// <summary>Probe volume evaluation mode.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:82]: The probe volume evaluation mode.
Can you also add:
\<seealso cref="ProbeVolumesEvaluationMode "/>

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 19]
```diff
@@ -51,32 +78,55 @@ public enum ShaderOptions
...
         ProbeVolumesEvaluationMode = ProbeVolumesEvaluationModes.Disabled,
+        /// <summary>Probe volume supports additive blending.</summary>
         ProbeVolumesAdditiveBlending = 1,
+        /// <summary>Probe volume filtering mode.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:86]: The probe volume filtering mode.
Can you also add:
\<seealso cref="ProbeVolumesEncodingModes"/>

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 20]
```diff
@@ -51,32 +78,55 @@ public enum ShaderOptions
...
         ProbeVolumesBilateralFilteringMode = ProbeVolumesBilateralFilteringModes.Validity,
+        /// <summary>Probe volume encoding.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:88]: The probe volume encoding method.
Can you also add:
<seealso cref="ProbeVolumesEncodingModes"/>

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 21]
```diff
@@ -51,32 +78,55 @@ public enum ShaderOptions
...
         AreaLights = 1,
 
+        /// <summary>Support for barn doors.</summary>
         BarnDoor = 0
     };
 
     // Note: #define can't be use in include file in C# so we chose this way to configure both C# and hlsl
     // Changing a value in this enum Config here require to regenerate the hlsl include and recompile C# and shaders
+    /// <summary>
+    /// Project wide shader configuration options.
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:101]: Project-wide shader configuration options.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 22]
```diff
@@ -51,32 +78,55 @@ public enum ShaderOptions
...
+    /// This class reflects the ShaderOptions enum. It's meant to be used in C# code to check current configuration.
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:102]: add this to a /// \<remarks> section please:
This class reflects the <see cref="ShaderOptions "/> enum. Use it in C# code to check the current configuration.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 23]
```diff
@@ -51,32 +78,55 @@ public enum ShaderOptions
...
     public class ShaderConfig
     {
-        public const int k_XRMaxViewsForCBuffer = 2;         // REALLY IMPORTANT! This needs to be the maximum possible XrMaxViews for any supported platform!
-                                                             // this needs to be constant and not vary like XrMaxViews does as it is used to generate the cbuffer declarations
+        // REALLY IMPORTANT! This needs to be the maximum possible XrMaxViews for any supported platform!
+        // this needs to be constant and not vary like XrMaxViews does as it is used to generate the cbuffer declarations
+        /// <summary>Maximum number of XR views for constant buffer allocation.</summary>
+        public const int k_XRMaxViewsForCBuffer = 2;
 
+        /// <summary>Use camera relative rendering to enhance precision.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:111]: Indicates whether to use [camera-relative rendering](../manual/Camera-Relative-Rendering.md) to enhance precision. A value of 1 enables camera-relative rendering and a value of 0 disables camera-relative rendering.

Comment (JordanL8) [reply com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:111]: For all of these below properties, can you add:  A value of x enables _feature_ and a value of 0 disables _feature_. Make it a [list](https://confluence.unity3d.com/pages/viewpage.action?spaceKey=DOCS&title=API+docs+content+and+formatting#sphnav93901911Packages) if there is more than one value.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 24]
```diff
@@ -51,32 +78,55 @@ public enum ShaderOptions
...
         public static int s_CameraRelativeRendering = (int)ShaderOptions.CameraRelativeRendering;
+        /// <summary>Use pre-exposition to enhance color precision.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:113]: Indicates whether to use pre-exposition to enhance color prevision.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 25]
```diff
@@ -51,32 +78,55 @@ public enum ShaderOptions
...
         public static int s_PreExposition = (int)ShaderOptions.PreExposition;
+        /// <summary>Maximum number of views for XR.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:115]: Indicates the maximum number of views to use for XR rendering.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 26]
```diff
@@ -51,32 +78,55 @@ public enum ShaderOptions
...
         public static int s_XrMaxViews = (int)ShaderOptions.XrMaxViews;
+        /// <summary>Precomputes atmospheric attenuation for the directional light on the CPU, which makes it independent from the fragment's position, which is faster but wrong.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:117]: Indicates whether to precompute atmosphere attenuation for the directional light on the CPU.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs (hunk 23 above)
Comment (JordanL8) [reply com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:111]: Also can you add a ///\<seealso cref=""/> with a  link to the shader options value? Just so we don't have to duplicate information.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 27]
```diff
@@ -51,32 +78,55 @@ public enum ShaderOptions
...
         public static int s_PrecomputedAtmosphericAttenuation = (int)ShaderOptions.PrecomputedAtmosphericAttenuation;
+        /// <summary>Probe volume evaluation mode.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:119]: Specifies the probe volume evaluation mode.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 28]
```diff
@@ -51,32 +78,55 @@ public enum ShaderOptions
...
         public static ProbeVolumesEvaluationModes s_ProbeVolumesEvaluationMode = (ProbeVolumesEvaluationModes)ShaderOptions.ProbeVolumesEvaluationMode;
+        /// <summary>Probe volume supports additive blending.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:121]: Indicates whether probe volumes support additive blending.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 29]
```diff
@@ -51,32 +78,55 @@ public enum ShaderOptions
...
         public static int s_ProbeVolumesAdditiveBlending = (int)ShaderOptions.ProbeVolumesAdditiveBlending;
+        /// <summary>Probe volume filtering mode.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:123]: Specifies the probe volume encoding method.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 30]
```diff
@@ -51,32 +78,55 @@ public enum ShaderOptions
...
         public static ProbeVolumesBilateralFilteringModes s_ProbeVolumesBilateralFilteringMode = (ProbeVolumesBilateralFilteringModes)ShaderOptions.ProbeVolumesBilateralFilteringMode;
+        /// <summary>Probe volume encoding.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:125]: Specifies the probe volume encoding method.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 31]
```diff
@@ -51,32 +78,55 @@ public enum ShaderOptions
...
         public static ProbeVolumesEncodingModes s_ProbeVolumesEncodingMode = (ProbeVolumesEncodingModes)ShaderOptions.ProbeVolumesEncodingMode;
+        /// <summary>Support for area lights.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:127]: Indicates whether to support area lights.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs [hunk 32]
```diff
@@ -51,32 +78,55 @@ public enum ShaderOptions
...
         public static int s_AreaLights = (int)ShaderOptions.AreaLights;
+        /// <summary>Support for barn doors.</summary>
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:129]: Indicates whether to support barn doors.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.LookDev.cs [hunk 33]
```diff
@@ -242,6 +242,10 @@ void IDataProvider.GetShadowMask(ref RenderTexture output, StageRuntimeInterface
             data.additionalCameraData.clearColorMode = oldClearMode;
         }
 
+        /// <summary>
+        /// Callback called to do any necessary cleanup.
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.LookDev.cs:246]: The callback HDRP raises to process any necessary cleanup.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/RenderPipelineSettings.cs [hunk 34]
```diff
@@ -157,6 +157,9 @@ internal static LightSettings NewDefault() => new LightSettings()
         }
 
 
+        /// <summary>
+        /// Resolution setting for planar reflections.
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/RenderPipelineSettings.cs:161]: Represents resolution settings for planar reflections.

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs (hunk 3 above)
Comment (JulienIgnace-Unity) [reply com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:15]: Some advanced stuff is a bit different but as it stands all this is very much experimental so one of the mode will probably go away anyway so I'm not sure we need to put more effort into this right now :)

File: com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs (hunk 6 above)
Comment (JulienIgnace-Unity) [reply com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs:27]: Spherical Harmonics are a mathematical construct used to represent spherical functions. The more level the more precision you have (and the more costly, both in memory and perf).

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.LookDev.cs (hunk 33 above)
Comment (JulienIgnace-Unity) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.LookDev.cs:246]: Actually here it's the HDRP implementation of the callback raised by the look dev.

Comment (JordanL8) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.LookDev.cs:246]: Ah okay, how about:
The HDRP implementation for the callback that the look dev raises to process any necessary cleanup.

**Review by JordanL8 — APPROVED** [review]

**Review by sebastienlagarde — APPROVED** [review]

**Pull request merged**